### PR TITLE
return 4xx errors instead of 500

### PR DESCRIPTION
**Pull request by boriwo**

### CODE REVIEW

File: internal/pkg/app/handlers_v1.go [hunk 1]
```diff
@@ -121,7 +120,7 @@ func (a *APIManager) addRouteHandler(w http.ResponseWriter, r *http.Request) {
 	err = a.routingTableMgr.AddRoute(ctx, &route)
 	if err != nil {
 		log.Ctx(ctx).Error().Str("op", "addRouteHandler").Msg(err.Error())
-		resp := ErrorResponse(err)
+		resp := ErrorResponse(&BadRequestError{err.Error(), err})
```
Comment (holidaymike) [inline internal/pkg/app/handlers_v1.go:123]: I don't think we can just do a catch-all here. There might be other errors such as db connections error that are not badrequest error. I think we need to either need to check the error returned by AddRoute or have AddRoute return different type of errors. Also, we really should cleanup the sentinel errors in routingTableMgr per our agreement

Comment (boriwo) [reply internal/pkg/app/handlers_v1.go:123]: so picky ;)

Comment (boriwo) [reply internal/pkg/app/handlers_v1.go:123]: on a more serious note i was also wondering if it really makes sense to raise 4xx (or 5xx) error codes from lower layers that are far away and detached from the rest api layer

Comment (holidaymike) [reply internal/pkg/app/handlers_v1.go:123]: Personally, I think it is OK to raise 4xx code from lower level to make our code more concise. Maybe we should have a bigger discussion

**Pull request closed without merging**